src/iOS/Xamarin.iOS/Samples/GeometryEngine/Buffer/Buffer.cs

```diff
@@ -48,124 +57,245 @@ public override void ViewDidLoad()
-                UIAlertController alertController = UIAlertController.Create("Geometry Engine Failed!", ex.Message, UIAlertControllerStyle.Alert);
+                // Display an error message if there is a problem generating the buffers.
+                UIAlertController alertController = UIAlertController.Create("Error creating buffers", ex.Message, UIAlertControllerStyle.Alert);
                 alertController.AddAction(UIAlertAction.Create("OK", UIAlertActionStyle.Default, null));
                 PresentViewController(alertController, true, null);
                 return;
             }
         }
 
+        private void ShowBufferSwatches(Colors planarBufferColor, Colors geodesicBufferColor)
+        {
+            // Create UIKit.UIColors to represent the System.Drawing.Colors used for the buffers.
+            UIColor planarLabelColor = UIColor.FromRGBA(planarBufferColor.R,
+                planarBufferColor.G,
+                planarBufferColor.B,
+                planarBufferColor.A);
+            UIColor geodesicLabelColor = UIColor.FromRGBA(geodesicBufferColor.R,
+                geodesicBufferColor.G,
+                geodesicBufferColor.B,
+                geodesicBufferColor.A);
+
+            // Show buffer symbol colors in the UI by setting the appropriate text view fill color.
+            _planarSwatchLabel.BackgroundColor = planarLabelColor;
+            _geodesicSwatchLabel.BackgroundColor = geodesicLabelColor;
+        }
+
+        private void ClearBuffersButton_TouchUpInside(object sender, EventArgs e)
+        {
+            // Clear the buffer and point graphics.
+            foreach (GraphicsOverlay ov in _myMapView.GraphicsOverlays)
+            {
+                ov.Graphics.Clear();
+            }
+        }
+
         private void CreateLayout()
         {
-            // Create the UILabel for instructions.
-            _bufferInstructionsUILabel = new UILabel();
-            _bufferInstructionsUILabel.Text = "Buffer (miles):";
-            _bufferInstructionsUILabel.AdjustsFontSizeToFitWidth = true;
-            _bufferInstructionsUILabel.BackgroundColor = UIColor.White;
-
-            // Create UITextFiled for the buffer value.
-            _bufferDistanceMilesUITextField = new UITextField();
-            _bufferDistanceMilesUITextField.Text = "10";
-            _bufferDistanceMilesUITextField.AdjustsFontSizeToFitWidth = true;
-            _bufferDistanceMilesUITextField.BackgroundColor = UIColor.White;
-            // - Allow pressing 'return' to dismiss the keyboard
-            _bufferDistanceMilesUITextField.ShouldReturn += (textField) => { textField.ResignFirstResponder(); return true; };
-
-            // Add the MapView and other controls to the page.
-            View.AddSubviews(_myMapView, _bufferInstructionsUILabel, _bufferDistanceMilesUITextField);
+            // Create the help label.
+            _helpLabel = new UILabel
```

Comment:
I recommend putting a UIToolbar between the help label as well, and also updating the toolbar at the bottom to be behind both rows of controls.